tests/PHPUnit/Unit/UrlTest.php

```diff
@@ -74,6 +75,17 @@ public function testGetCurrentHost($description, $test)
         $this->assertEquals($test[4], Url::getCurrentHost(), $description);
     }
 
+    public function testGetHostWithTrustedHosts()
+    {
+        Config::getInstance()->General['enable_trusted_host_check'] = 1;
+        Access::getInstance()->setSuperUserAccess(true);
+        $_SERVER['HTTP_HOST'] = null;
+
+        // plugins/CoreAdminHome/API::setTrustedHosts();
+        Url::saveTrustedHostnameInConfig(['example.com','stats.example.com']);
```

Comment:
```suggestion
        Config::getInstance()->General['enable_trusted_host_check'] = 1;
        Config::getInstance()->General['trusted_hosts'] = ['example.com','stats.example.com'];
        $_SERVER['HTTP_HOST'] = null;

```

There are already several tests setting the config value directly instead of using an API for it.